### PR TITLE
[CARBONDATA-3771] Fixed date filter issue and use hive constants instead of hard-coded values

### PR DESCRIPTION
**Pull request by kunal642**
### Why is this PR needed?
Filter on date column is giving wrong results due to incorrect carbon expression. 
 
 ### What changes were proposed in this PR?
For date column getExprString is giving the filter value as 'DATE2020-01-01' which causes the filter to fail.
Changed to `((ExprNodeConstantDesc)` `exprNodeDesc).getValue().toString()` for ConstantDesc for better handling
    
 ### Does this PR introduce any user interface change?
 - No
 - Yes. (please explain the change and update document)

 ### Is any new testcase added?
 - No
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2717/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1005/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2718/

File: core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java [hunk 1]
```diff
@@ -320,6 +321,28 @@ public static boolean writeSegmentFileForOthers(
...
+    CarbonFile[] indexFiles = segmentFolder.listFiles(new CarbonFileFilter() {
+      @Override
+      public boolean accept(CarbonFile file) {
+        return (file.getName().endsWith(CarbonTablePath.INDEX_FILE_EXT) || file.getName()
+            .endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT));
+      }
+    });
+    String segmentFileName = genSegmentFileName(segmentId, UUID) + CarbonTablePath.SEGMENT_EXT;
+    try {
+      SegmentFileStore sfs = new SegmentFileStore(tablePath, segmentFileName);
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java:338]: once SegmentFileStore is made, call `sfs.getIndexFiles()`, we can remove list files of line 329

File: integration/hive/src/main/java/org/apache/carbondata/hive/Hive2CarbonExpression.java [hunk 2]
```diff
@@ -103,38 +111,43 @@ public static Expression convertExprHive2Carbon(ExprNodeDesc exprNodeDesc) {
...
-            new LiteralExpression(l1.get(right).getExprString().replace("'", ""),
+            new LiteralExpression(getExpressionValue(l1.get(right)),
                 getDateType(l1.get(right).getTypeString()));
         return new EqualToExpression(columnExpression, literalExpression);
       } else if (udf instanceof GenericUDFOPEqualOrGreaterThan) {
         ColumnExpression columnExpression = new ColumnExpression(l1.get(left).getCols().get(left),
             getDateType(l1.get(left).getTypeString()));
-        LiteralExpression literalExpression = new LiteralExpression(l1.get(right).getExprString(),
-            getDateType(l1.get(left).getTypeString()));
+        LiteralExpression literalExpression =
```
Comment (ajantha-bhat) [inline integration/hive/src/main/java/org/apache/carbondata/hive/Hive2CarbonExpression.java:120]: can we make two methods ?
`makeColumnExpression(l1)
``makeLiteralExpression(l1)`

I see lot of duplicate code here

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonOutputCommitter.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  @Override
+  public void commitJob(JobContext jobContext) throws IOException {
+    try {
+      Configuration configuration = jobContext.getConfiguration();
+      CarbonLoadModel carbonLoadModel = MapredCarbonOutputFormat.getLoadModel(configuration);
+      ThreadLocalSessionInfo.unsetAll();
+      SegmentFileStore.writeSegmentFile(carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable(),
+          carbonLoadModel.getSegmentId(), String.valueOf(carbonLoadModel.getFactTimeStamp()));
+      SegmentFileStore
+          .mergeIndexAndWriteSegmentFile(carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable(),
```
Comment (ajantha-bhat) [inline integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonOutputCommitter.java:113]: So, for nonntransactional also we write segment files now ?

**Comment by ajantha-bhat** [comment]
please update the PR description

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1006/

File: integration/hive/src/main/java/org/apache/carbondata/hive/Hive2CarbonExpression.java (hunk 2 above)
Comment (kunal642) [reply integration/hive/src/main/java/org/apache/carbondata/hive/Hive2CarbonExpression.java:120]: i dont think its good, we would just be replacing 1 line code with a method call.

File: core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java (hunk 1 above)
Comment (kunal642) [reply core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java:338]: this is a static method inside SegmentFileStore

Comment (kunal642) [reply core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java:329]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1012/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2724/

**Comment by ajantha-bhat** [comment]
LGTM

**Pull request closed without merging**